extract_all_binaries.js

```diff
@@ -130,33 +130,54 @@ var getWindowsRedisFiles = function() {
 
 var downloadAndExtractTGZ = function(filepath, destDir, agentCoreFlag) {
  	if (agentCoreFlag) {
-		fs.createReadStream('binaries/agentcore/tgz/'+filepath).pipe(zlib.createGunzip()).on('error', function(err) {
-			console.log('ERROR: Failed to gunzip ' + filepath + ': ' + err.message);
-			process.exit(1);
-		})
-		.pipe(tar.Extract({path: destDir})).on('error', function(err) {
-			console.log('ERROR: Failed to untar ' + filepath + ': ' + err.message);
-			process.exit(1);
-		})
-		.on('close', function() {
-			console.log('Download and extract of ' + filepath + ' finished.');
-		});
+    if(fs.existsSync('binaries/agentcore/tgz/'+filepath)) {
+		  fs.createReadStream('binaries/agentcore/tgz/'+filepath).pipe(zlib.createGunzip()).on('error', function(err) {
+			  console.log('ERROR: Failed to gunzip ' + filepath + ': ' + err.message);
+        tryRebuild();
+		  })
+		  .pipe(tar.Extract({path: destDir})).on('error', function(err) {
+			  console.log('ERROR: Failed to untar ' + filepath + ': ' + err.message);
+        tryRebuild();
+		  })
+		  .on('close', function() {
+			  console.log('Download and extract of ' + filepath + ' finished.');
+		  });
+    } else {
+      console.log(filepath + " does not exist.")
+      tryRebuild();
+    }
 	} else {
-		fs.createReadStream('binaries/appmetrics/tgz/'+filepath).pipe(zlib.createGunzip()).on('error', function(err) {
-			console.log('ERROR: Failed to gunzip ' + filepath + ': ' + err.message);
-			process.exit(1);
-		})
-		.pipe(tar.Extract({path: destDir})).on('error', function(err) {
-			console.log('ERROR: Failed to untar ' + filepath + ': ' + err.message);
-			process.exit(1);
-		})
-		.on('close', function() {
-			console.log('Download and extract of ' + filepath + ' finished.');
-		});
+    if(fs.existsSync('binaries/appmetrics/tgz/'+filepath)) {
+		  fs.createReadStream('binaries/appmetrics/tgz/'+filepath).pipe(zlib.createGunzip()).on('error', function(err) {
+			  console.log('ERROR: Failed to gunzip ' + filepath + ': ' + err.message);
+        tryRebuild();
+		  })
+		  .pipe(tar.Extract({path: destDir})).on('error', function(err) {
+			  console.log('ERROR: Failed to untar ' + filepath + ': ' + err.message);
+        tryRebuild();
+		  })
+		  .on('close', function() {
+			  console.log('Download and extract of ' + filepath + ' finished.');
+		  });
+    } else {
+      console.log(filepath + " does not exist.")
+      tryRebuild();
+    }
 	}
 	
 };
 
+function tryRebuild() {
+  console.log('Falling back to node-gyp rebuild');
+  var childProcess = require('child_process')
+  var res = childProcess.spawnSync('node-gyp', ['rebuild'], {stdio:'inherit', shell:true});
+  if(res.signal) {
```

Comment:
What happens when the parent process receives a signal instead? Does it properly terminate the child process on all platforms? @sam-github you might know the answer to this since I know you've done a deep dive in to this before..